src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
+    public MyLinkedList() {
+        firstNode = null;
+        lastNode = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
```

Comment:
create a private method for this, and use where needed
```suggestion
      checkIndex(index);
```